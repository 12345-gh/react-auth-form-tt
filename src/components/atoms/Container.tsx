import React from "react";
import styled from "styled-components";

interface ContainerProps {
    children: React.ReactNode;
}

const StyledContainer = styled.div`
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    height: 100vh;
    padding: 20px;
    box-sizing: border-box;
    background: linear-gradient(167.96deg, #f4f9ff 0%, #e0edfb 100%);
`;

export const Container: React.FC<ContainerProps> = ({ children }) => {
    return <StyledContainer>{children}</StyledContainer>;
};
