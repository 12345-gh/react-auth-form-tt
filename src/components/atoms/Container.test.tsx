import { render, screen } from "@testing-library/react";
import { describe, expect, it } from "vitest";

import { Container } from "./Container";

describe("Container component", () => {
    it("matches the snapshot", () => {
        const { asFragment } = render(
            <Container>
                <div>Child Element</div>
            </Container>,
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders children correctly", () => {
        render(
            <Container>
                <div>Child Element</div>
            </Container>,
        );
        expect(screen.getByText("Child Element")).toBeInTheDocument();
    });
});
