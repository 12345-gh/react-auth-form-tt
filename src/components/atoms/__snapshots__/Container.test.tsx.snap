// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-braxZu SzWhb"
  >
    <div>
      Child Element
    </div>
  </div>
</DocumentFragment>
`;
